Assets/Managers/pipeManager.cs

```diff
@@ -20,39 +21,88 @@ void Start()
     // Update is called once per frame
     void Update()
     {
-        
+        if(Input.GetKeyDown(KeyCode.Q)){
+            beginTrans();
+        }
     }
 
     void beginTrans(){
-        Vector2 foundStartChange = new Vector2(0,0);
-        bool result = tranverse(foundStartChange,startPos);
-        if(result == true){
+        succCon = false;
+        Vector2 foundStart = gridControl.returnStartPosition();
+        Debug.Log("Starting Trans");
+        tranverse(Vector2.right,foundStart);
+        if(succCon == true){
             Debug.Log("YAAAAS U WIN");
+        }else{
+            Debug.Log("Failed End");
         }
     }
 
-    bool tranverse(Vector2 change, Vector2 currentPos){
+    void tranverse(Vector2 change, Vector2 currentPos){
+        Vector2 oldPos = currentPos;
+        Debug.Log($"old Postion: {oldPos}");
         currentPos = new Vector2((int)(currentPos.x + change.x),(int)(currentPos.y + change.y));
-        GameObject nextPiece = gridControl.returnBoardObject(currentPos);
-        if(nextPiece == null){
-            return false;
-        }else if (nextPiece.tag == "end"){
-            return true;
-        }else if (nextPiece.tag == "pipe"){
+        Debug.Log($"New Postion: {currentPos}");
+        GameObject currentPiece = gridControl.returnBoardObject(currentPos);
+        if(currentPiece == null){
+            Debug.Log("YUUUP its null");
+            return;
+        }else if (currentPiece.gameObject.tag == "end"){
+            succCon = true;
+            return;
+        }else if (currentPiece.gameObject.tag == "pipe"){
             //get stuff off object and call transverse again
-            Vector2[] nextPoints = nextPiece.GetComponent<pipe>().returnPipeDirections();
-            bool tempResult = false;
-            foreach(Vector2 element in nextPoints){
-                bool result = tranverse(element,currentPos);
-                if(result){
-                    tempResult = true;
+            Debug.Log("we in here!");
+            Vector2[] connectingPoints = currentPiece.GetComponent<pipe>().returnPipeDirections();
+
+
+            ///Validating pipe connection section of algorithim
+            Vector2 connectingPostion = Vector2.zero; // init
+            foreach(Vector2 pointSet in connectingPoints){
+                if (new Vector2((int)(currentPos.x + pointSet.x),(int)(currentPos.y + pointSet.y)) == oldPos){
+
+                    connectingPostion = pointSet;// new Vector2((int)(currentPos.x + pointSet.x),(int)(currentPos.y + pointSet.y));
+                }
+            }
+
+            //This section deals with transerving out all possible ends of the pipe that is not the entry side
+            foreach(Vector2 element in connectingPoints){
```

Comment:
I'm happy with the progress made on this algorithm. 
Need to remember to remove the Debug.Logs. 
Note: only works with bendy bois and straight bois.